lib/new_post_manager.rb

```diff
@@ -82,7 +82,18 @@ def self.post_needs_approval?(manager)
     is_fast_typer?(manager) ||
     matches_auto_silence_regex?(manager) ||
     WordWatcher.new("#{manager.args[:title]} #{manager.args[:raw]}").requires_approval? ||
-    (SiteSetting.approve_unless_staged && user.staged)
+    (SiteSetting.approve_unless_staged && user.staged) ||
+    post_needs_approval_in_its_category?(manager)
+  end
+
+  def self.post_needs_approval_in_its_category?(manager)
+    if manager.args[:topic_id].present?
+      Category.joins(:topics).find_by(topics: { id: manager.args[:topic_id] }).require_reply_approval?
+    elsif manager.args[:category].present?
+      Category.find(manager.args[:category]).require_topic_approval?
```

Comment:
this is weird ... is category misnamed here and should be category_id ?

Reply:
I was also super confused.

https://github.com/discourse/discourse/blob/22c4000375ce23d9ad1be8fb444ea008b156884f/app/assets/javascripts/discourse/models/composer.js.es6#L30-L35

https://github.com/discourse/discourse/blob/ee1eb1a5bd834ed49e7efe2ee58bb567556a8c7c/app/controllers/posts_controller.rb#L597-L602

I'll _try_ to rename this in this PR, unless it causes much havoc 😅 and I will open another PR.